packages/@aws-cdk/cdk/lib/util/physical-name-generator.ts

```diff
@@ -0,0 +1,58 @@
+import crypto = require('crypto');
+import { IResource } from "../resource";
+import { Stack } from '../stack';
+
+export class PhysicalNameGenerator {
+  public static from(resource: IResource): PhysicalNameGenerator {
+    return new PhysicalNameGenerator(
```

Comment:
This is not good enough. `node.id` is not unique. 
I think we should able to simply use `node.uniqueId`. It's doing all the heavy lifting already.

Reply:
Can we keep it this way for now...? I don't really feel like defining 3 more interfaces for these. Remember that this class is private to the `@aws-cdk/cdk` package.

Reply:
Done.